core/aws-core/src/main/java/software/amazon/awssdk/awscore/exception/AwsServiceException.java

```diff
@@ -68,17 +69,22 @@ public String getMessage() {
             if (extendedRequestId() != null) {
                 details.add("Extended Request ID: " + extendedRequestId());
             }
-            String message = super.getMessage();
             if (message == null) {
                 message = awsErrorDetails().errorMessage();
             }
             if (message == null) {
                 return details.toString();
             }
-            return message + " " + details;
+            StringBuilder formattedMessage = new StringBuilder(message);
```

Comment:
Can we add a comment stating that appending the attempt information after the details is done to preserve the order, ensuring that AWS error details are printed after the AWS server exception message?